raysort/sort_utils.py

```diff
@@ -301,15 +301,15 @@ def validate_part(cfg: AppConfig, pinfo: PartInfo) -> Tuple[int, bytes]:
 
 
 def compare_checksums(input_checksums: List[str], output_summary: str) -> bool:
-    input_checksum = sum(input_checksums)
-    input_checksum = str(hex(input_checksum))[2:][-16:]
-
     assert "Checksum: " in output_summary, output_summary
-    checksum_line = output_summary.split("Checksum: ")[1][:16]
+    checksum_line = output_summary.split("Checksum: ")[1]
     output_checksum = checksum_line.split()[0]
+    input_checksum = sum(input_checksums)
+    input_checksum = str(hex(input_checksum))[2:]
```

Comment:
pro tip: `f"{input_checksum:x}"` gives you the hex representation without the `0x` prefix.